src/main/java/com/selfxdsd/storage/SelfContracts.java

```diff
@@ -289,4 +300,28 @@ private Contract buildContract(final Record rec){
             rec.getValue(SLF_CONTRACTS_XDSD.ROLE),
             this.storage);
     }
+
+    /**
+     * Transaction Result compatible.
+     * Transactions apparently are not working well with jooq MySQL dialect on
+     * testing H2.
+     * @param jooq DSLContext.
+     * @param callable TransactionalCallable.
+     * @param <R> Result Type.
+     * @return Result.
+     * @throws Throwable If something went wrong and rollback is triggered.
+     * @checkstyle IllegalThrows (10 lines).
+     */
+    private <R> R transactionResultCompat(
+        final DSLContext jooq,
+        final TransactionalCallable<R> callable
+    ) throws Throwable {
+        final R result;
+        if (this.database.dbms().equals(Database.Dbms.H2)) {
+            result = callable.run(jooq.configuration());
+        } else {
+            result = jooq.transactionResult(callable);
```

Comment:
@criske I don't really understand the difference between ``callable.run(jooq)`` and ``jooq.transactionResult(callable)``. Isn't it the same dialect in both cases? How is the problem fixed? :D

Reply:
@amihaiemil the `callabale#run` is not running in a transaction. Is just like calling calling `Runnable#run()` instead of passing it to a thread to do it.